server/wal/wal.go

```diff
@@ -797,11 +800,25 @@ func (w *WAL) sync() error {
 		return nil
 	}
 
+	if w.openshiftWarnFsyncDuration == nil {
+		defaultWarnFsyncDuration := warnSyncDuration
+		w.openshiftWarnFsyncDuration = &defaultWarnFsyncDuration
+		if warnFsyncDurationOverride := os.Getenv("OPENSHIFT_WARN_FSYNC_DURATION"); warnFsyncDurationOverride != "" {
+			override, err := strconv.Atoi(warnFsyncDurationOverride)
+			if err != nil {
+				w.lg.Sugar().Infof("OPENSHIFT_WARN_FSYNC_DURATION specified but could not be parsed. falling back to default of %v. parse error: %v", warnSyncDuration, err)
+			} else {
+				openshiftWarnFsyncDuration := time.Duration(override) * time.Second
+				w.openshiftWarnFsyncDuration = &openshiftWarnFsyncDuration
+			}
+		}
+	}
```

Comment:
This was put in place to make the duration in which an fdatasync is deemed "too slow" configurable as we are already making other timeouts/thresholds configurable. I did this because I saw flakes with this present after the original request timeout change, but realize this may not actually be _necessary_ if we are OK with the occasional flake here.

Reply:
This change was strictly for changing the threshold for the fsync duration. If we feel this isn't necessary I think it is reasonable to revert this particular set of logic

Reply:
I think it's useful .. if we get it properly configured and through upstream. I expect the 3.6 release right around the corner and it would be nice to not add things we immediately have to patch again.

Reply:
That's fair. I'm happy to pursue something upstream, but I imagine I won't get around to it within the timeline I'm imagining the 3.6 release to fall in.

I _do_ think we are likely going to want to keep the static delay increase and logic in place though. How painful is it to carry this patch until something is done upstream?

Reply:
Probably painful for both avenues, for the same reasons cited above. 
Does the request timeout function above help in our CI or did this turn out to be a red herring?

Reply:
I don't think it resolved the problem at hand for the bug so in that sense it could be considered a red herring.

IIRC it was mentioned that there was a _slight_ improvement in the regressions in 4.19 after this change went in, but not enough to get rid of the regression the bug was filed for.

If it makes the most sense to revert this change, I won't stop that from happening.

If we revert I would like to at least file another bug (or something like that) to track that we should probably pursue making a change upstream (or re-apply this patch in an easier way to maintain long term) to track our desire to give etcd a longer request timeout period to align with our kube-apiserver max timeout of 34s.

Reply:
Because I'm not really familiar with the processes and how decisions for something like reverting this are made, I'll defer to you on whether or not this should be reverted.

If it should be reverted, I should have some time this afternoon to push up a PR for the revert. If this is urgent to revert, and you have time before I will to do so, please feel free to go ahead with it.

Reply:
thanks for the clarification, @everettraven :) 
I think @dusk125 is back today, I'll hash this out with him. Don't spend anymore time on this, thanks for your help.